ts/features/messages/components/MessageAttachments.tsx

```diff
@@ -24,130 +18,52 @@ type MessageAttachmentProps = {
   attachment: UIAttachment;
 } & PartialProps;
 
-type MessageAttachmentsProps = {
-  attachments: ReadonlyArray<UIAttachment>;
-} & PartialProps;
-
-const styles = StyleSheet.create({
-  opacityDisabled: {
-    opacity: 0.35
-  },
-  opacityEnabled: {
-    opacity: 1.0
-  },
-  container: {
-    paddingRight: 0,
-    paddingLeft: 0,
-    marginVertical: 0,
-    height: 80,
-    backgroundColor: IOColors.white
-  },
-  row: {
-    flexDirection: "row",
-    alignItems: "center",
-    justifyContent: "space-between",
-    flex: 1
-  },
-  icon: {
-    alignSelf: "center"
-  },
-  middleSection: {
-    flex: 1,
-    paddingLeft: 8,
-    alignSelf: "center"
-  }
-});
+type MessageAttachmentsProps = WithTestID<
+  {
+    attachments: ReadonlyArray<UIAttachment>;
+  } & PartialProps
+>;
 
-/**
- * Custom icon for the attachment, based on the contentType (at the moment only the pdf have a custom representation)
- * @param props
- * @constructor
- */
-const AttachmentIcon = (props: {
-  contentType: UIAttachment["contentType"];
-}) => {
-  switch (props.contentType) {
+const getFormatByContentType = (
+  contentType: UIAttachment["contentType"]
+): ModuleAttachmentProps["format"] => {
+  switch (contentType) {
     case ContentTypeValues.applicationPdf:
-      return <Icon name="docAttachPDF" color="blue" size={32} />;
+      return "pdf";
     default:
-      return <Icon name="docAttach" color="blue" size={32} />;
+      return "doc";
   }
 };
 
-/**
- * Represent a single attachment.
- * An item that displays the file name and size.
- * @param props
- * @constructor
- */
-const AttachmentItem = (props: MessageAttachmentProps) => {
+const AttachmentItem = ({
+  attachment,
+  openPreview,
+  downloadAttachmentBeforePreview
+}: MessageAttachmentProps) => {
   const { downloadPot, onAttachmentSelect } = useAttachmentDownload(
-    props.attachment,
-    props.downloadAttachmentBeforePreview,
-    props.openPreview
+    attachment,
+    downloadAttachmentBeforePreview,
+    openPreview
   );
+
   return (
-    <TouchableOpacity
-      style={[
-        styles.container,
-        props.disabled ? styles.opacityDisabled : styles.opacityEnabled
-      ]}
+    <ModuleAttachment
+      testID="message-attachment"
```

Comment:
This prevents a test to be able to differentiate between different attachments. What do you think about either providing the testId from the outside (using props) or finding a way to generate an unique and easy retrievable testId?

Reply:
In the next [PR](https://github.com/pagopa/io-app/pull/5161), since `AttachmentItem` is not exported, I added a testID on the parent component to differentiate the various attachments. What do you think?

Reply:
In such case, should we remove this testId property on the internal un-exported component?

Reply:
it's useful if we want to just test whether _n_ items are rendered

``` Typescript
const f24Container = getByTestId("f24-list-container");
const attachments = within(f24Container).queryAllByTestId("message-attachment");
expect(attachments.length).toBe(n);
```

Reply:
alternatively, we can remove the testID property on the internal un-exported component and use `queryAllByA11yRole` to retrieve all elements

``` Typescript
const f24Container = getByTestId("f24-list-container");
const attachments = within(f24Container).queryAllByA11yRole("button");
expect(attachments.length).toBe(n);
```
What do you think?